src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java

```diff
@@ -215,13 +261,77 @@ else if (targetY <= levelDepth && workerY > levelDepth)
                 //First calculate way to miner building.
                 newProxy = getProxy(buildingPos, worker.getPosition(), BlockPosUtil.getDistanceSquared(worker.getPosition(), buildingPos));
 
+
                 //Then add the ladder position as the latest node.
-                proxyList.add(new BlockPos(ladderPos.getX(), level.getDepth(), ladderPos.getZ()));
+                proxyList.add(
+                  new BlockPos(
+                                ladderPos.getX() + building.getVectorX() * OTHER_SIDE_OF_SHAFT,
+                          level.getDepth(),
+                          ladderPos.getZ() + building.getVectorZ() * OTHER_SIDE_OF_SHAFT));
+
+                if(level.getRandomNode() != null && level.getRandomNode().getParent() != null)
+                {
+                    final List<BlockPos> nodesToTarget = new ArrayList<>();
+                    com.minecolonies.coremod.entity.ai.citizen.miner.Node currentNode = level.getNode(level.getRandomNode().getParent());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this fully qualified name with "Node" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1942)